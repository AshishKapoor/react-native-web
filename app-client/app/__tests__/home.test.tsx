import React from "react";
import { create, type ReactTestRendererJSON } from "react-test-renderer";
import Home from "../index";
import { TamaguiProvider } from "tamagui";
import config from "../../tamagui.config";

jest.mock("expo-router", () => ({ Drawer: { Screen: "Home" } }));

describe("Home screen", () => {
  const HomeComponent = (
    <TamaguiProvider config={config}>
      <Home />
    </TamaguiProvider>
  );

  it("renders home screen correctly", () => {
    const home = create(HomeComponent).toJSON() as ReactTestRendererJSON;
    const layout = home.children![0] as ReactTestRendererJSON;
    const screen = layout.children![0] as ReactTestRendererJSON;

    expect(home.type).toBe("View");
    expect(layout.type).toBe("Text");
    expect(layout?.props?.testID).toBe("empty-text");
    expect(screen).toBe("Empty Screen");
  });
});
