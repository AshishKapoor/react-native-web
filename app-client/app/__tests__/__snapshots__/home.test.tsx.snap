// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
    testID="empty-text"
  >
    Empty Screen
  </Text>
</View>
`;
